qiskit/pulse/builder.py

```diff
@@ -286,62 +294,74 @@ def __init__(self,
         Args:
             backend (Union[Backend, BaseBackend]): Input backend to use in
                 builder. If not set certain functionality will be unavailable.
-            schedule: Initital schedule to build on. If not supplied
-                a schedule will be created.
-            name: Name of pulse program to be built. Only used if `schedule`
-                is not ``None``.
-            default_alignment: Default scheduling alignment policy for the
-                builder. One of 'left', 'right', 'sequential', or an alignment
-                contextmanager.
+            block: Initital ``ScheduleBlock`` to build on.
+            name: Name of pulse program to be built.
+            default_alignment: Default scheduling alignment for builder.
+                One of ``left``, ``right``, ``sequential`` or an instance of
+                :class:`~qiskit.pulse.transforms.alignments.AlignmentKind` subclass.
             default_transpiler_settings: Default settings for the transpiler.
             default_circuit_scheduler_settings: Default settings for the
                 circuit to pulse scheduler.
+
+        Raises:
+            PulseError: When invalid ``default_alignment`` or `block` is specified.
         """
         #: BaseBackend: Backend instance for context builder.
         self._backend = backend
 
         #: Union[None, ContextVar]: Token for this ``_PulseBuilder``'s ``ContextVar``.
         self._backend_ctx_token = None
 
-        #: pulse.Schedule: Active schedule of BuilderContext.
-        self._context_schedule = None
-
         #: QuantumCircuit: Lazily constructed quantum circuit
         self._lazy_circuit = None
 
-        # ContextManager: Default alignment context.
-        self._default_alignment_context = _align(default_alignment)
+        #: Dict[str, Any]: Transpiler setting dictionary.
+        self._transpiler_settings = default_transpiler_settings or dict()
+
+        #: Dict[str, Any]: Scheduler setting dictionary.
+        self._circuit_scheduler_settings = default_circuit_scheduler_settings or dict()
 
-        self._transpiler_settings = default_transpiler_settings or {}
+        #: List[ScheduleBlock]: Stack of context.
+        self._context_stack = []
 
-        if default_circuit_scheduler_settings is None:
-            default_circuit_scheduler_settings = {}
-        self._circuit_scheduler_settings = \
-            default_circuit_scheduler_settings or {}
+        #: str: Name of the output program
+        self._name = name
 
-        # pulse.Schedule: Root program context-schedule
-        self._schedule = schedule or Schedule(name=name)
+        if block is not None:
+            if not isinstance(block, ScheduleBlock):
+                raise exceptions.PulseError(f'Input program {repr(block)} is not `ScheduleBlock`. '
+                                            'This cannot be a root context.')
```

Comment:
Couldn't a schedule be trivially made into a `ScheduleBlock`? (This looks like a breaking change. If it's not, then it's safe to ignore this comment)

Reply:
Good question. The docstring of `pulse.build` says
```
        schedule: A *mutable* pulse ``ScheduleBlock`` in which your pulse program will
            be built.
```
I guess this indicates 
```python
my_sched1 = Schedule()
with build(schedule=my_sched1):
    # do something

my_sched2 = Schedule()
with build() new_sched:
    # do something
my_sched1 == my_sched2.append(new_sched)
```

So the builder should operate on the input `schedule` in the mutable fashion. However we cannot append `ScheduleBlock` to `Schedule` because of the ambiguity of duration. Thus this will become breaking change. If there is any approach to avoid this, I'm happy to update the logic.

Reply:
This is what worked before:
```
with build(schedule=Schedule(...)) as new_schedule:
   ...
```
it looks like this would error now. My suggestion is to do something like:
```
if isinstance(block, Schedule):
    tmp = ScheduleBlock()
    tmp.append(block)  # I forget what the interface is here. 'call'?
    block = tmp
```

Reply:
Okey, so it seems like the docstring is not correct. `block` is not necessary to be *mutable*?